### PR TITLE
Add support for print-scaling option.

### LINKED ISSUE #108: Support "print-scaling" option
The "print-scaling" option is the standard IPP Everywhere option for controlling the scaling of the output; currently defined values are:

- `auto`: if the “ipp-attribute-fidelity” attribute is true or the document is larger than the requested media, scale the document using the 'fit' method if the margins are non- zero, otherwise scale using the 'fill' method. If the “ipp-attribute-fidelity” attribute is false or unspecified and the document is smaller than the requested media, scale using the 'none' method.
- `auto-fit`: if the “ipp-attribute-fidelity” attribute is true or the document is larger than the requested media, scale the document using the ‘fit’ method. Otherwise, scale using the ‘none’ method.
- `fill`: scale the document to fill the requested media size, preserving the aspect ratio of the document data but potentially cropping portions of the document.
- `fit`: scale the document to fit the printable area of the requested media size, preserving the aspect ratio of the document data without cropping the document.
- `none`: do not scale the document to fit the requested media size. If the document is larger than the requested media, center and clip the resulting output. If the document is smaller than the requested media, center the resulting output.

This attribute is defined in [PWG 5100.16: IPP Transaction-Based Printing Extensions](https://ftp.pwg.org/pub/pwg/candidates/cs-ipptrans10-20131108-5100.16.pdf).

### PR DESCRIPTION
**Pull request by dheeraj135**
Support for the print-scaling feature has been added to imagetoraster, imagetopdf and pdftopdf filters.
Fixes #108.

### CODE REVIEW

File: filter/imagetopdf.c [hunk 1]
```diff
@@ -997,6 +958,113 @@ main(int  argc,				/* I - Number of command-line arguments */
   colorspace = ColorDevice ? CUPS_IMAGE_RGB_CMYK : CUPS_IMAGE_WHITE;
 
   img = cupsImageOpen(filename, colorspace, CUPS_IMAGE_WHITE, sat, hue, NULL);
+  if(img!=NULL){
+
+  int margin_defined = 0;
+  int fidelity = 0;
+  int document_large = 0;
+
+  if(ppd->custom_margins[0]||ppd->custom_margins[1]||ppd->custom_margins[2]||ppd->custom_margins[3])
```
Comment (dheeraj135) [inline filter/imagetopdf.c]: @tillkamppeter Is this the right way to check for margins?

Comment (tillkamppeter) [reply filter/imagetopdf.c]: I think not, AFAIR the custom_margins[] array is for the margins for a custom page size. If as output page size one of the list of the PageSize option is chosen, the margins assigned to this size (from the ImageableArea lines in the PPD, should be also somewhere in CUPS' ppd data structure) are the ones you are looking for.

Comment (dheeraj135) [reply filter/imagetopdf.c]: Thanks, @tillkamppeter. I have updated the PR.

**Comment by tillkamppeter** [comment]
Dheeraj, thank you verymuch for your contribution. I have merged it now so that it will make part of the 1.23.0 release of cups-filters.

**Pull request merged**